kedro-datasets/kedro_datasets/geopandas/geojson_dataset.py

```diff
@@ -18,6 +19,13 @@
     get_protocol_and_path,
 )
 
+from kedro_datasets.pandas.generic_dataset import (
+    NON_FILE_SYSTEM_TARGETS as PANDAS_NON_FILE_SYSTEM_TARGETS,
+)
+
+# Unofficially GeoPandas also supports pandas io-methods, such as read_csv.
+NON_FILE_SYSTEM_TARGETS = ["postgis"] + PANDAS_NON_FILE_SYSTEM_TARGETS
+
```

Comment:
I don't like this too much, but I don't know if there are better ways.

Officially just documented `geopandas read_*` methods should be supported, but `geopandas.read_csv` also works because geopandas extends pandas...

This tries to rescue the panda's non-file system targets but also adds a dependency to another dataset...

Reply:
What happens if one reads a non-geo file with GeoPandas? Does it return a normal pandas DataFrame?

Reply:
I tried to write a test for this, somehow it didn't worked anymore. Nevermind, I cleaned up the code...